src/lcm/create_params_template.py

```diff
@@ -1,34 +1,31 @@
 """Create a parameter template for a model specification."""
 
 import inspect
-from typing import Any
 
 import jax.numpy as jnp
 import pandas as pd
 from jax import Array
 
+from lcm.model import Model
+from lcm.typing import Params
+
 
 def create_params_template(
-    model_spec: dict[str, Any],
+    user_model: Model,
     variable_info: pd.DataFrame,
     grids: dict[str, Array],
-    default_params: dict[str, float] | None = None,
-) -> dict[str, dict[str, Any] | float]:
+    default_params: dict[str, int | float] | None = None,
+) -> Params:
```

Comment:
This makes me think that some params Object comes out here; I would prefer to see `PyTree` even if that is less precise.

Reply:
`ParamsDict`